File: uniquePaths.py
def uniquePaths(m,n):
	dp=[[1]*n for _ in range(m)]

	for row in range(1,m):
		for col in range(1,n):
			dp[row][col]=dp[row-1][col]+dp[row][col-1]

	return dp[m-1][n-1]


print(uniquePaths(3,7))

File: uniquePaths.java
// Approach 1- DP (Bottom- Up) 
import java.util.*;
class Solution{ // TC O(M*N) SC O(M*N)
  public int uniquePaths(int m, int n) {
    int[][] d = new int[m][n];

    for(int[] arr : d) {
      Arrays.fill(arr, 1);
    }
    for(int col = 1; col < m; ++col) {
      for(int row = 1; row < n; ++row) {
        d[col][row] = d[col - 1][row] + d[col][row - 1];
      }
    }
    return d[m - 1][n - 1];
  }
}

class uniquePaths{
  public static void main(String[] args) {
    int m=3;
    int n=7; 
    Solution s= new Solution();
    System.out.println(s.uniquePaths(m,n));
  }
}